src/sagas/RootSaga.js

```diff
@@ -1,138 +1,110 @@
-  education: 'levelOfEducation',
-  socialLinks: socialLinks => socialLinks.filter(({ socialLink }) => socialLink !== null),
-};
-
-export const mapDataForRequest = (props) => {
-  const state = {};
-  Object.keys(props).forEach((prop) => {
-    const propModifier = PROP_TO_STATE_MAP[prop] || prop;
-    if (typeof propModifier === 'function') {
-      state[prop] = propModifier(props[prop]);
-    } else {
-      state[propModifier] = props[prop];
-    }
-  });
-  return state;
-};
-
-
-export function* saveUserProfile(action) {
-  const { username, userAccountState } = action.payload;
+import * as ProfileApiService from '../services/ProfileApiService';
+
+export function* handleFetchProfile(action) {
+  const { username } = action.payload;
 
   try {
-    yield put(saveUserProfileBegin());
+    yield put(fetchProfileBegin());
+    const profile = yield call(
+      ProfileApiService.getProfile,
+      username,
+    );
 
-    // Use call([context, fnName], ...args) for proper context on apiService
-    const userAccount = yield call(
-      [userAccountApiService, 'saveUserAccount'],
+    yield put(fetchProfileSuccess(profile));
+    yield put(fetchProfileReset());
+  } catch (e) {
+    yield put(fetchProfileFailure(e.message));
+  }
+}
+
+export function* handleSaveProfile(action) {
+  const { username, userAccountState } = action.payload;
+  try {
+    yield put(saveProfileBegin());
+    const profile = yield call(
+      ProfileApiService.patchProfile,
       username,
-      mapDataForRequest(userAccountState),
+      userAccountState,
     );
 
     // Tells the profile form that
-    yield put(saveUserProfileSuccess());
-    // TODO: export the fetchUserAccountSuccess action from frontend-auth so we can
-    // dry this up.
-    yield put({
-      type: 'FETCH_USER_ACCOUNT_SUCCESS',
-      payload: { userAccount },
-    });
+    yield put(saveProfileSuccess());
+    yield put(fetchProfileSuccess(profile));
     yield delay(300);
     yield put(closeEditableField(action.payload.fieldName));
     yield delay(300);
-    yield put(saveUserProfileReset());
+    yield put(saveProfileReset());
   } catch (e) {
-    yield put(saveUserProfileFailure(e));
+    yield put(saveProfileFailure(e.message));
   }
 }
 
-
-export function* saveUserProfilePhoto(action) {
+export function* handleSaveProfilePhoto(action) {
   const { username, formData } = action.payload;
 
   try {
-    yield put(saveUserProfilePhotoBegin());
+    yield put(saveProfilePhotoBegin());
 
     // Use call([context, fnName], ...args) for proper context on apiService
-    yield call([userAccountApiService, 'saveUserProfilePhoto'], username, formData);
+    yield call(ProfileApiService.postProfilePhoto, username, formData);
 
     // Get the account data. Saving doesn't return anything on success.
-    const userAccount = yield call(
-      [userAccountApiService, 'getUserAccount'],
-      username,
-    );
+    yield handleFetchProfile(fetchProfileAction);
```

Comment:
This can delegate this part to the whole fetch profile saga, which is nice and DRYer.